hnn_core/dipole.py

```diff
@@ -58,6 +62,12 @@ def simulate_dipole(net, n_trials=None, record_vsoma=False):
         raise TypeError("record_vsoma must be bool, got %s"
                         % type(record_vsoma).__name__)
 
+    if record_isoma is not None and isinstance(record_isoma, bool):
+        net.params['record_isoma'] = record_isoma
```

Comment:
is there such a parameter in the `params` file? If not, let's not invent it ...

Reply:
So this was posed as the workaround for enabling/disabling `vsoma` recordings. Do you think this should be an attribute of the `Network` instead?

Reply:
My vote is to leave `record_isoma` as an argument of `simulate_dipole()` without creating a new param. Same with `record_vsoma`, however, since it's already an argument in the default param file as well as the GUI (right?), I guess it's fine to leave `record_vsoma` in there until after GUI integration. I'm on an anti-params-file kick at the moment so that we can hopefully refocus and simplify the use-case of parameters stored in param files :)

Reply:
Ah I missed this. `save_vsoma` is contained inside `default.json` but I think that's just leftover from copying the GUI params directly. 

Currently this is just an awkward way to pass `record_vsoma` to the `network_builder.py`. The params variable is just a convenient way to pass these variables, but neither is specified in the actual `.params` file. 

I'm not a fan of adding extra arguments to `_BACKEND.simulate(net)` so the alternative would be adding an attribute to `Network` which can be accessed by `network_builder.py` later on.

Reply:
we have a bit of a mess with parallel backends. Maybe you get away with `net.params` now. But we'll have to bite the bullet at some point and clean it up. If we don't touch the code, we won't realize how to improve it :)

Reply:
Oh I see. Yeah that will be a bit of a mess, but I feel like `_BACKEND.simulate(net)` is the appropriate place for arguments that determine the types of data that are output/saved in a simulation. Any specific arguments against this?

Reply:
Fair point, my only hesitancy is tacking on a bunch of arguments to the critical functions. More of a clean/readability concern, but these seem sufficiently important.

Reply:
Decided to go with making them attributes of `Network` since they can be updated once and readout by `network_builder.py`. Realized that passing them as arguments would require a lot of copy-pasted code as I tried it.

Reply:
do you mind leaving a comment where these parameters are used again? I guess in `cell.py` or `_simulate_single_trial`?